common/src/main/java/io/netty/util/internal/Hidden.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.util.internal;
+
+import io.netty.util.concurrent.FastThreadLocalThread;
+import reactor.blockhound.BlockHound;
+import reactor.blockhound.integration.BlockHoundIntegration;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Contains classes that must be have public visibility but are not public API.
+ */
+class Hidden {
+
+    /**
+     * This class integrates Netty with BlockHound.
+     *
+     * It is public but only because of the ServiceLoader's limitations
+     * and SHOULD NOT be considered a public API.
+     */
+    @UnstableApi
+    public static class NettyBlockHoundIntegration implements BlockHoundIntegration {
+
+        @Override
+        public void applyTo(BlockHound.Builder builder) {
+            builder.nonBlockingThreadPredicate(new Function<Predicate<Thread>, Predicate<Thread>>() {
+                @Override
+                public Predicate<Thread> apply(final Predicate<Thread> p) {
+                    return new Predicate<Thread>() {
+                        @Override
+                        public boolean test(Thread thread) {
+                            return p.test(thread) || thread instanceof FastThreadLocalThread;
+                        }
+                    };
+                }
+            });
+        }
+
+        @Override
+        public int compareTo(BlockHoundIntegration o) {
+            return 0;
```

Comment:
Add a proper impl ?

Reply:
the default method `compareTo` in `BlockHoundIntegration` (can't use it due to Java 6 compiler, had to copy) returns `0` by default, because the order of most of integrations does not matter.
And, in Netty, since it does not depend on any other integration, it is so.
Or would you recommend returning some other constant (`+/-1`)?

Reply:
I decided to crowdsource it a bit :D 
https://twitter.com/bsideup/status/1186220275232321537

Although I see why you're suggesting the 0/-1 option. Just want to double check (and maybe change the default we have in BlockHound)

Reply:
Ok, after a discussion with Mr. Shipilёv, it seems that `0` is a better choice here, to avoid a situation when `a < b` and `b < a` at the same time. WDYT?

Reply:
I liked the solution proposed in https://twitter.com/tagir_valeev/status/1186270999383330818 which won't guarantee a consistent ordering over restarts, but is a total ordering for the lifetime of the process

Reply:
I guess any non-constant based solution should be done [in the default method](https://github.com/reactor/BlockHound/blob/f0064db55a0754bab0ca2858a0fb6d47262b0169/agent/src/main/java/reactor/blockhound/integration/BlockHoundIntegration.java#L42) defined in BlockHound itself, to keep it consistent.
Hereby I suggest leaving `0` here as it is not the responsibility of this integration.
WDYT?